src/Components/Modal/Modal.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import './Modal.css';
+import cancelIcon from "../../Images/cancel_icon.png";
+
+
+export default class Modal extends React.Component{
+    onClose = (e)=> {
+        this.props.onClose && this.props.onClose(e);
+    };
+
+    render () {
+        if (!this.props.show){
+            return null;
+        }
+        return (
+            <div className="Modal">
+                <div className="content">
+                    {this.props.children}
+                    <div className="header">
+                        <img className="annuler"
+                             src={cancelIcon}
+                             onClick={(e) => {this.onClose(e)}} alt="Annuler"
+                        />
+                    </div>
+                </div>
+            </div>
+        );
+    }
```

Comment:
Tu peux transformer cette classe en un functional component
Pour résoudre le problème de ce fait de `onClose`, tu peux faire directement `onClick=this.props.onClose` et rajouter si nécessaire dans le parent un `if (this.state.show)` pour remplacer `this.props.onClose`

Reply:
Done